core/base/db-upgrades-manager.php

```diff
@@ -212,12 +214,32 @@ public function get_upgrade_callbacks() {
 				continue;
 			}
 
+			if ( ! $this->should_run( $method_name ) ) {
+				continue;
+			}
+
 			$callbacks[] = [ $upgrades_class, $method_name ];
 		}
 
 		return $callbacks;
 	}
 
+	public function should_run( $method_name ) {
+		$upgrades_conditions_class = $this->get_upgrades_conditions_class();
+		$upgrades_consditions_reflection = new \ReflectionClass( $upgrades_conditions_class );
+
+		$condition_methods = $upgrades_consditions_reflection->getMethods();
+
+		if ( ! in_array( $method_name, $condition_methods ) ) {
+			return true; // There is no condition.
+		}
+
+		$method_name = $method_name . $this->should_run_postfix;
+		$should_run = call_user_func( [ $upgrades_conditions_class, $method_name ] );
```

Comment:
What about this?

```suggestion
		if ( in_array( $method_name, $condition_methods ) ) {
			$method_name = $method_name . $this->should_run_postfix;
		 	return $should_run = call_user_func( [ $upgrades_conditions_class, $method_name ] );	
		} else {
			return true;     
		}
```
no negative condition...
I think it is more readable.